### ISSUE TITLE
launch error

### ISSUE THREAD

**Issue by colinallen** [inpho/topic-explorer#221]
When doing topicexplorer launch on a mac, the hostname function returns <name>.local for some user-specified name, but the browser can't resolve <name>.local:8000.

Both localhost:8000 and 127.0.0.1:8000 would work on Mac

We need to test whether both of these work in Windows 
(and linux, but I'm fairly sure the answer to that is 'yes').

Some evidence that only 127.0.0.1 will work on Windows: 
https://superuser.com/questions/713491/why-does-localhost-not-work-in-windows-8-1-but-127-0-0-1-does

Assuming that 127.0.0.1:<portnumber> works on all platforms, then I propose that this should be hard coded into the launch script, instead of using whatever 'hostname' returns.

**Comment by kirtansakariya** [comment]
Both of those work for windows

**Comment by colinallen** [comment]
Ok — so is there any reason (Jaimie?) not to alter the script to launch to localhost:8000 instead of hostname:8000 ? 

Even in cases when installing a public server, one will launch --nobrowser, so it shouldn’t make a difference there.

**Comment by colinallen** [comment]
Also found that the existing approach fails when connected via vpn and hostname returns something like 'vpn-128-194-199-115.tamulink.tamu.edu' (yes, I'm on Texas A&M vpn).

So, we should switch to hardcoded 'localhost' or '127.0.0.1'

**Comment by kirtansakariya** [comment]
We tested two options, one was to not rewrite 0.0.0.0 (line 602 in server.py) and the other is to hard code a rewrite from 0.0.0.0 to localhost (line 602 in server.py). First option allows us to delete 3 lines of code. Second option does not allow removal. Is there a reason to prefer the second option over the first one?
See this for discussion: https://github.com/Pylons/pyramid/issues/489

**Comment by JaimieMurdock** [comment]
`0.0.0.0` tells the server to listen on all hostnames, changing to `localhost` means that the server would only listen to requests made to `localhost` (and *not* `127.0.0.1`, `compy386`, or `jaimie-masked-his-ip.washdc.fios.verizon.net`). So the server should stay at 0.0.0.0 with the rewrite. 

I would do the hardcode rewrite to `localhost` or `127.0.0.1` so the URL printed to the console and opened in the browser is right.

I originally changed the behavior due to some behavior at an IU lab where `localhost` access was explicitly blocked or caused firewall issues that the `socket.gethostname()` didn't cause. If we run into problems like that again, perhaps we can come up with a how-to-fix guide using the `topicexplorer launch config.ini --host URLHOST` flag.

tl;dr -- hardcode rewrite from `0.0.0.0` to `localhost`.

**Comment by kirtansakariya** [comment]
it works on mac, will test on windows later tonight

**Comment by kirtansakariya** [comment]
localhost works on windows as well, just tested it.

**Comment by JaimieMurdock** [comment]
Merged the branch
